xarray/tests/test_convert.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+import xarray as xr
+import iris
+from iris.coords import DimCoord
+from iris.cube import Cube
+
+
+def test_from_iris_no_var_name():
+    latitude = DimCoord(np.linspace(-90, 90, 4),
+                        standard_name='latitude',
+                        units='degrees')
+    cube = Cube(np.zeros((4,), np.float32),
+                dim_coords_and_dims=[(latitude, 0)])
+    xr.DataArray.from_iris(cube)
```

Comment:
Ideally, we would using `xarray.testing.assert_identical()` to compare the resulting DataArray from this operation to one that you construct directly. That's a more stringent test than merely verifying that these don't crash.